src/features/snippets/logic/filter.ts

```
@@ -0,0 +1,31 @@
+
+function filterLineLength(snippet: string): boolean {
+  const lines = snippet.split("\n");
+  return lines.every((line) => line.trim().length <= MAX_LINE_LENGTH);
+}
+
+function filterSnippetLength(snippet: string): boolean {
+  return snippet.length >= MIN_SNIPPET_LENGTH && snippet.length <= MAX_SNIPPET_LENGTH;
+}
+
+function filterSnippetSpecialCharacters(snippet: string): boolean {
+  const regex = /^[\x00-\x7F]+$/;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)